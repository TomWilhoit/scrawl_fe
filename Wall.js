import React from "react";
import {
  Text,
  View,
  TouchableOpacity,
  TextInput,
  ScrollView
} from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";
var Filter = require("bad-words");

export default class Wall extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      key: null,
      text: "",
      comments: null
    };
  }

  componentDidMount = () => {
    this.gatherComments();
    setInterval(() => this.gatherComments(), 1000);

  };

  gatherComments = async () => {
    const pk = this.props.currentWall.pk;
    const url = `https://scrawlr.herokuapp.com/api/v1/walls/${pk}`;
    try {
      const response = await fetch(url);
      let responseJson = await response.json();
      const reversed = responseJson.comments.reverse();
      const reversedTwice = reversed.reverse()
      this.setState({
        comments: reversedTwice
      });
    } catch (error) {
      console.log(error);
    }
  };

  onSubmit = async () => {
    const check = this.props.checkProximity(
      this.props.currentWall.lat,
      this.props.currentWall.lng
    );
    if (check) {
      const newComment = this.state.text;
      if (newComment.length > 0) {
        const options = {
          method: "POST",
          headers: {
            "Content-Type": "application/JSON"
          },
          body: JSON.stringify({ comment: newComment })
        };
        this.setState({
          comments: [...this.state.comments, newComment],
          text: ''
        });
        const pk = this.props.currentWall.pk;
        const url = `https://scrawlr.herokuapp.com/api/v1/walls/${pk}/comments`;
        try {
          const response = await fetch(url, options);
          let responseJson = await response.json();
        } catch (error) {
          console.log(error);
        }
      } else {
        alert("Type more?");
      }
    } else {
      alert("YOU DON'T EVEN GO HERE");
    }
  };

  displayComments = () => {
    const check = this.props.checkProximity(
      this.props.currentWall.lat,
      this.props.currentWall.lng
    );
    console.log('Wall Check method', check)
    if (check) {
      if(this.state.comments === null){
        return (
          <Text
            style={{
              marginTop: 130,
              textAlign: "center",
              fontWeight: "bold",
              color: "#eaeaea",
              fontSize: 20
            }}
          >
            LOADING COMMENTS
          </Text>
        );
    }else if (this.state.comments.length > 0) {
        filter = new Filter();
        const commentsDisplay = this.state.comments.map((comment, index) => {
          let cleanComment = filter.clean(comment);
          return (
            <Text
              style={{
                paddingLeft: 10,
                margin: 5,
                fontSize: 22.5,
                color: "#27476e"
              }}
              key={index}
            >
              {cleanComment}
            </Text>
          );
        });
        return commentsDisplay;
      } else if(this.state.comments.length === 0) {
        return (
          <Text
            style={{
              marginTop: 130,
              textAlign: "center",
              fontWeight: "bold",
              color: "#eaeaea",
              fontSize: 20
            }}
          >
            No Comments Yet!
          </Text>
        );
      }
    } else {
      return <Text
      style={{
        marginTop: 130,
        textAlign: "center",
        fontWeight: "bold",
        color: "#eaeaea",
        fontSize: 20
      }}
    >
      You are too far from this wall. 
    </Text>;
    }
  };

  render() {
    return (
      <View
        style={{
          flex: 1,
          flexDirection: "column",
          backgroundColor: "#006992"
        }}
      >
        <TouchableOpacity
          style={{
            height: 35,
            marginTop: 25,
            justifyContent: "center",
            paddingLeft: 15
          }}
          onPress={() => this.props.onPress("home")}
          textAlign="left"
        >
          <Icon name="angle-left" size={30} color="white" />
        </TouchableOpacity>
        <View
          style={{
            justifyContent: "center",
            alignItems: "center",
            marginTop: 35
          }}
        >
          {this.props.fontLoaded ? (
            <Text
              style={{
                fontFamily: "PermanentMarker",
                fontSize: 30,
                color: "#f7f9f9",
                fontSize: 30
              }}
            >
              {this.props.currentWall.name}
            </Text>
          ) : null}
        </View>
        <TextInput
          style={{
            height: 60,
            backgroundColor: "white",
            width: "100%",
            textAlign: "center",
            justifyContent: "center"
          }}
          value={this.state.text}
          placeholder="Say something, anything bruh it's your world."
          onChangeText={text => this.setState({ text })}
        />
        <ScrollView
          style={{
            maxHeight: 400,
            backgroundColor: "lightblue",
            width: "100%",
            stickyBottom: true
          }}
        >
          {this.displayComments()}
        </ScrollView>
        <TouchableOpacity
          style={{
            height: 100,
            backgroundColor: "#87b38d",
            width: "100%",
            justifyContent: "center",
            marginBottom: -57
          }}
          onPress={() => this.onSubmit("home")}
        >
          <Text
            style={{
              textAlign: "center",
              fontWeight: "bold",
              color: "white",
              fontSize: 40
            }}
          >
            Submit
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}
