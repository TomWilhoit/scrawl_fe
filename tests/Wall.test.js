import React from 'react';
import { shallow } from "enzyme";
import Wall from '../Wall';


describe("Wall", () => {
  let wrapper;
  const mockFunc = jest.fn()
  const mockWall = {
    latitude: 39.7653,
    longitude: -104.9791,
    title: "Improper City",
    subtitle: "Subtitle"
  }
  
  beforeEach(() => {
    wrapper = shallow(<Wall currentWall={mockWall} />);
  });

  it('should match the snapshot when on the Home page', () => {
    expect(wrapper).toMatchSnapshot();
  });

})