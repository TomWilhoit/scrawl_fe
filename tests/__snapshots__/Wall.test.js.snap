// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wall should match the snapshot when on the Home page 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#006992",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "height": 35,
        "justifyContent": "center",
        "marginTop": 25,
        "paddingLeft": 15,
      }
    }
    textAlign="left"
  >
    <Icon
      allowFontScaling={false}
      color="white"
      name="angle-left"
      size={30}
    />
  </TouchableOpacity>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 35,
      }
    }
  />
  <ScrollView
    style={
      Object {
        "backgroundColor": "lightblue",
        "maxHeight": 400,
        "stickyBottom": true,
        "width": "100%",
      }
    }
  >
    <Component>
      You cannot reach this wall. Turn back while you can.
    </Component>
  </ScrollView>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Say something, anything bruh it's your world."
    style={
      Object {
        "backgroundColor": "white",
        "height": 60,
        "justifyContent": "center",
        "textAlign": "center",
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#87b38d",
        "height": 100,
        "justifyContent": "center",
        "marginBottom": -57,
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "color": "white",
          "fontSize": 40,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Submit
    </Component>
  </TouchableOpacity>
</Component>
`;
